### DISCUSSION TITLE
3D Masking doesn't work in Unity

### DISCUSSION THREAD

**Discussion by ArvisMag** [naverz/zepeto-studio-global#23] [General]
Hellow! I am following the Zepeto tutorial about "3D - Rigging & Masking" and I can't figure out why it is not working for me. I did as the tutorial is describing in Maya and in Blender -> at the end result is the same, with no difference in Unity wen prevuing my 3D model. It doesn't hide some body parts that have vertexes painted black.

What can be, that I am doing wrong?

![screen_03](https://user-images.githubusercontent.com/6276317/178455743-91e7e723-9a5e-47ab-9a18-008881d0f6a7.PNG)

![screen_02](https://user-images.githubusercontent.com/6276317/178455754-0f654673-4301-4297-9397-58ae16417598.PNG)

![screen_01](https://user-images.githubusercontent.com/6276317/178455757-27681995-4041-42f8-a247-5e0d1547d8f1.PNG)

**Comment by naverz-mkao** [comment]
Hi ArvisMag, 

Looking at your screen shots, it looks like you did everything correctly on the exporting side. Could you first verify that the vertex painting has actually been exported correctly into Unity, then we can go from there?

Please try the steps below and post a screen shot of your result.



**1: Drag or import your character fbx file into the unity assets folder.**

<img width="909" alt="image" src="https://user-images.githubusercontent.com/107931355/180488827-5f3086c6-de68-41fd-af94-1947e9f7672f.png">

**2: Click on the little arrow to expand the prefab.**

<img width="844" alt="image" src="https://user-images.githubusercontent.com/107931355/180491898-ba15d13f-e5c2-46fa-917b-2c9969c06a04.png">

**3: Click on the "mask" mesh and look in your inspector to see your mask mesh.**

<img width="1406" alt="image" src="https://user-images.githubusercontent.com/107931355/180492250-4350beb1-3f1d-490f-ba29-4164dec33d56.png">

**4: Change this dropdown to the "Vertex Color" option.**

<img width="1390" alt="image" src="https://user-images.githubusercontent.com/107931355/180492416-5d588ee1-ed1a-45c5-af5c-16e0ff1f7e39.png">

**5: Do you see your model with vertex colors in black and do you see the bottom text saying "UV1, Color"?**

<img width="610" alt="image" src="https://user-images.githubusercontent.com/107931355/180492765-4c328e3f-657d-4e50-9bb9-e9ac69d5c02f.png">

**Please post your result as an image**

If you do not get the result like in the image above where your vertex color shows up black on the "mask" mesh, and the bottom does not say "UV1, Color" then there could be some weird setting in your Maya or Blender file. If that is the case, you can try copy pasting the contents of your blender file into a fresh file and try re-exporting.

Let's do this first test then go from here!
